charts/atlantis/templates/statefulset.yaml

```
@@ -522,7 +525,15 @@ spec:
             mountPath: /home/atlantis/.netrc
             subPath: netrc
           {{- end }}
-          {{- if or .Values.aws.credentials .Values.aws.config .Values.awsSecretName }}
+          {{- range $filename, $_ := .Values.aws }}
+          {{- if has $filename (list "credentials" "config") }}
+          - name: aws-volume
+            readOnly: true
+            mountPath: {{ $.Values.aws.directory | default "/home/atlantis/.aws" }}/{{ $filename }}
+            subPath: {{ $filename }}
+          {{- end }}
+          {{- end }}
+          {{- if .Values.awsSecretName }}
```

I would just run this if the previous is empty, otherwise it will append both and will likely fail.
```suggestion
          {{- else if .Values.awsSecretName }}
```

---

This is a bit tricky due to the if statement being outside the range block.  I can add another if at the top, but I'm ultimately not sure what the intent is when specifying awsSecretName.  Is that intended to be mutually exclusive with the use of aws.config?  As right now if I specify both(in my own local copy), the volumeMounts get created based on the .aws.X filenames, but the volume references awsSecretName and the secret doesnt get created with that name.

---

Yeah, the logic is not right (lint failed). An easy way is to give priority to the existing secret since it has to be created externally in before, so requires extra planning:

```tpl
{{- if .Values.awsSecretName }}
- name: aws-volume
readOnly: true
mountPath: {{ .Values.aws.directory | default "/home/atlantis/.aws" }}
{{- else }}
{{- range $filename, $_ := .Values.aws }}
{{- if has $filename (list "credentials" "config") }}
- name: aws-volume
readOnly: true
mountPath: {{ $.Values.aws.directory | default "/home/atlantis/.aws" }}/{{ $filename }}
subPath: {{ $filename }}
{{- end }}
{{- end }}
{{- end }}
```

If you decide to use this logic then a comment on values.yaml saying it has priority should be enough. This is also compatible with current implementation. WDYT?